### ISSUE TITLE
[Bug?]Session Hang during training with 'mnist_replica.py' and learning rate as a placeholder

### ISSUE THREAD

**Issue by hongjic** [tensorflow/tensorflow#11958]
### System information
 OS: Mac OS 
 TF: 1.2.0, pip install
 Python: 2.7.9
also tested on RHEL with TF 1.1 and Python 2.7.13, the problem still exists. I don't think it is system specific.

### Describe the problem
I am training the 'mnist-replica.py' provided on github. link is [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/tools/dist_test/python/mnist_replica.py) 
and set up two workers and one parameter server on my laptop.
I made only one modification to the code:
 use learning_rate as a placeholder rather than a fixed value and feed 0.01 to it each step.
When I train it asynchronously, it is ok to go, but the session hangs in .run() after one or two steps when I train it in synchronous.

Can anyone figure out the reason of this?

### Source code / logs
the source code is 
[here](https://drive.google.com/open?id=0Bw4fA7bI0IScTVlsT3FidS0tMXM). just provide it for ease. It is almost the same as the original one.

**Comment by hongjic** [comment]
I am not sure is this the right way to do?
